lib/Net/Async/Blockchain/ETH.pm

```diff
@@ -264,7 +264,10 @@ you received.
 async sub recursive_search {
     my ($self) = @_;
 
-    return undef unless $self->base_block_number;
+    unless ($self->base_block_number) {
+        warn("base_block_number is empty");
```

Comment:
@arshad-binary I think we can ignore a message that  `base block is empty` as it will not be helpful to investigate further either we add proper information or can remove the warn log.

Reply:
there are very less chances this will happen but i just added in case it happened.

Reply:
please remove this warn message, that is no issues in have the empty base_block_number, this is an optional parameter and we should not print warn messages for it.